src/env/node/pathMapping/sharedGKDataFolder.ts

```diff
@@ -1,73 +1,118 @@
 import os from 'os';
 import path from 'path';
 import { Uri, workspace } from 'vscode';
+import { xdgData } from 'xdg-basedir';
 import { Logger } from '../../../system/logger';
 import { wait } from '../../../system/promise';
 import { getPlatform } from '../platform';
 
-export const sharedGKDataFolder = '.gk';
+/** @deprecated prefer using XDG paths */
+const legacySharedGKDataFolder = path.join(os.homedir(), '.gk');
 
-export async function acquireSharedFolderWriteLock(): Promise<boolean> {
-	const lockFileUri = getSharedLockFileUri();
+class SharedGKDataFolderMapper {
+	private _initPromise: Promise<void> | undefined;
+	constructor(
+		// do soft migration, use new folders only for new users (without existing folders)
+		// eslint-disable-next-line @typescript-eslint/no-deprecated
+		private sharedGKDataFolder = legacySharedGKDataFolder,
+		private _isInitialized: boolean = false,
+	) {}
 
-	let stat;
-	while (true) {
+	private async _initialize() {
+		if (this._initPromise) {
+			throw new Error('cannot be initialized multiple times');
+		}
 		try {
-			stat = await workspace.fs.stat(lockFileUri);
+			await workspace.fs.stat(Uri.file(this.sharedGKDataFolder));
 		} catch {
-			// File does not exist, so we can safely create it
-			break;
+			// Path does not exist, so we can safely use xdg paths it
+			if (xdgData) {
+				this.sharedGKDataFolder = path.join(xdgData, 'gk');
+			} else {
+				this.sharedGKDataFolder = path.join(os.homedir());
```

Comment:
@nzaytsev 
It looks like it's an assignment to `$HOME`. Should it be `$HOME/.gk`?

Reply:
yes, thank you